xarray/plot/plot.py

```diff
@@ -218,66 +304,23 @@ def line(darray, *args, **kwargs):
     xincrease = kwargs.pop('xincrease', True)
     yincrease = kwargs.pop('yincrease', True)
     add_legend = kwargs.pop('add_legend', True)
+    _labels = kwargs.pop('_labels', True)
 
     ax = get_axis(figsize, size, aspect, ax)
-
-    error_msg = ('must be either None or one of ({0:s})'
-                 .format(', '.join([repr(dd) for dd in darray.dims])))
-
-    if x is not None and x not in darray.dims:
-        raise ValueError('x ' + error_msg)
-
-    if y is not None and y not in darray.dims:
-        raise ValueError('y ' + error_msg)
-
-    if x is not None and y is not None:
-        raise ValueError('You cannot specify both x and y kwargs'
-                         'for line plots.')
-
-    if ndims == 1:
-        dim, = darray.dims  # get the only dimension name
-
-        if (x is None and y is None) or x == dim:
-            xplt = darray.coords[dim]
-            yplt = darray
-
-        else:
-            yplt = darray.coords[dim]
-            xplt = darray
-
-    else:
-        if x is None and y is None and hue is None:
-            raise ValueError('For 2D inputs, please specify either hue or x.')
-
-        if y is None:
-            xlabel, huelabel = _infer_xy_labels(darray=darray, x=x, y=hue)
-            ylabel = darray.name
-            xplt = darray.coords[xlabel]
-            yplt = darray.transpose(xlabel, huelabel)
-
-        else:
-            ylabel, huelabel = _infer_xy_labels(darray=darray, x=y, y=hue)
-            xlabel = darray.name
-            xplt = darray.transpose(ylabel, huelabel)
-            yplt = darray.coords[ylabel]
-
-        huecoords = darray[huelabel]
-        huelabel = label_from_attrs(huecoords)
-
-    xlabel = label_from_attrs(xplt)
-    ylabel = label_from_attrs(yplt)
+    xplt, yplt, xlabel, ylabel, huelabel = _infer_line_data(darray, x, y, hue)
 
     _ensure_plottable(xplt)
 
     primitive = ax.plot(xplt, yplt, *args, **kwargs)
 
-    if xlabel is not None:
-        ax.set_xlabel(xlabel)
+    if _labels:
+        if xlabel is not None:
+            ax.set_xlabel(xlabel)
 
-    if ylabel is not None:
-        ax.set_ylabel(ylabel)
+        if ylabel is not None:
+            ax.set_ylabel(ylabel)
 
-    ax.set_title(darray._title_for_slice())
+        ax.set_title(darray._title_for_slice())
 
     if darray.ndim == 2 and add_legend:
         ax.legend(handles=primitive,
```

Comment:
`huecoords` on the line below this one is no longer defined -- that's why your tests are failing

Reply:
Ya, the merge is a little screwed. Here's the commit you need to fix it: https://github.com/dcherian/xarray/commit/3e0429cb3b43db8e74ece681e6b1d2f5f2b638d9. 

@yohai I can't push to your branch so you need to make those changes somehow. Here's a gist with a patch file. Download and apply it using `git apply` (https://www.devroom.io/2009/10/26/how-to-create-and-apply-a-patch-with-git/)